CRM/ACL/BAO/ACL.php

```diff
@@ -530,22 +530,37 @@ private static function loadDenyIDs(int $contactID, string $tableName, int $type
     $aclKeys = array_keys($acls);
     $aclKeys = implode(',', $aclKeys);
     $query = "
-SELECT   a.operation, a.object_id
+SELECT   a.operation, a.object_id, a.weight
   FROM   civicrm_acl_cache c, civicrm_acl a
  WHERE   c.acl_id       =  a.id
    AND   a.is_active    =  1
    AND   a.object_table = %1
    AND   a.id        IN ( $aclKeys )
    AND   a.deny         = 1
-GROUP BY a.operation,a.object_id
+GROUP BY a.operation,a.object_id, a.weight
```

Comment:
Why are you grouping by `weight`? I would think you would want to order by weight instead.